File: 274.h-指数.py
#
# @lc app=leetcode.cn id=274 lang=python3
#
# [274] H 指数
#

# @lc code=start
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        # 计数排序
        n = len(citations)
        counter = [0 for _ in range(n+1)]
        for citation in citations:
            counter[min(citation,n)] += 1
        h = n
        s = counter[n]
        while h > s:
            h -= 1
            s += counter[h]
        return h
# @lc code=end

File: 274.h-指数.cpp
/*
 * @lc app=leetcode.cn id=274 lang=cpp
 *
 * [274] H 指数
 */
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
// @lc code=start
class Solution
{
public:
    int hIndex(vector<int> &citations)
    {
        bool flag = true;
        int h = 0;
        int n = citations.size();
        while (flag == true)
        {
            int count = 0;
            for (int i = 0; i < n; i++)
            {
                if (citations[i] >= h)
                {
                    count++;
                }
            }
            if (count >= h)
            {
                h++;
            }
            else
            {
                flag = false;
            }
        }
        return h - 1;
    };
};
// @lc code=end
